web3/utils/address.py

```diff
@@ -0,0 +1,44 @@
+from eth_typing import (
+    ChecksumAddress,
+    HexAddress,
+)
+from eth_utils import (
+    keccak,
+    to_checksum_address,
+)
+from rlp import (
+    encode,
+)
+
+from web3.types import (
+    HexStr,
+    Nonce,
+)
+
+
+def get_create_address(sender: HexAddress, nonce: Nonce) -> ChecksumAddress:
+    """
+    Determine the resulting `CREATE` opcode contract address for a sender and a nonce.
+    """
+    bytes_sender = bytes.fromhex(sender[2:])
+
+    contract_address = keccak(encode([bytes_sender, nonce])).hex()[-40:]
```

Comment:
nit: Ideally we import `rlp` and use `rlp.encode` here for clarification. We've done the same with `import abi` -> `abi.encode()` and it just makes it easier to read the code.

```suggestion
    contract_address = keccak(rlp.encode([bytes_sender, nonce])).hex()[-40:]
```

Reply:
It's more readable as `rlp.encode`, agreed!